### ISSUE TITLE
'str' object is not callable during show_prediction (xgboost 0.7)

### ISSUE THREAD

**Issue by sudarshan1413** [TeamHG-Memex/eli5#252]
Hi,

I was trying to execute the exact code mentioned at eli5 documentation over here..

xgboost tutorial : http://eli5.readthedocs.io/en/latest/tutorials/xgboost-titanic.html

but during below step...I am getting an error..
Code:
from eli5 import show_prediction
show_prediction(clf, valid_xs[1], vec=vec, show_feature_values=True)

Error Message:
![image](https://user-images.githubusercontent.com/17721958/34735400-53dec4f8-f595-11e7-9e09-85f9f0832461.png)

Let me know if I am missing anything over here...

My Configuration :- Windows 7, Python 2.7.13 

Any help would be really appreciated !

**Comment by lopuhin** [comment]
Thanks for reporting this @sudarshan1413

This is an issue with recently released xgboost 0.7, eli5 currently does not support it.

**Comment by sudarshan1413** [comment]
Thanks @lopuhin for your quick response...!  Any workaround to solve this.... Will backdated (previous version) xgboost work?  If yes which specific version..

**Comment by lopuhin** [comment]
@sudarshan1413 the last version supported is ``pip install xgboost==0.6a2``, ``pip install 'xgboost<0.7'`` gives the same

**Comment by sudarshan1413** [comment]
Thanks @lopuhin . I will give it a try!

**Comment by antoinemercier82** [comment]
@lopuhin Hi Konstantin, the problem is that it is, at least for me, already quite tricky to install xgboost 0.7 on a Windows machine. I tried to uninstall xgboost 0.7 and install 0.6a2 but with no success. Any chance you would have another workaround?

Thanks

**Comment by beeva-josealberto** [comment]
This works for me in xgboost version 0.7.post3:

`booster = model.get_booster()`
`eli5.show_prediction(estimator = booster, doc = test_set.iloc[0,:])`

Hope it helps!

**Comment by ianozsvald** [comment]
I have a workaround, I have the same problem and I've monkey-patched the function from inside a Jupyter Notebook cell:

```
import eli5

def _check_booster_args(xgb, is_regression=None):
    # type: (Any, bool) -> Tuple[Booster, bool]
    if isinstance(xgb, eli5.xgboost.Booster): # patch (from "xgb, Booster")
        booster = xgb
    else:
        booster = xgb.get_booster() # patch (from "xgb.booster()" where `booster` is now a string)
        _is_regression = isinstance(xgb, XGBRegressor)
        if is_regression is not None and is_regression != _is_regression:
            raise ValueError(
                'Inconsistent is_regression={} passed. '
                'You don\'t have to pass it when using scikit-learn API'
                .format(is_regression))
        is_regression = _is_regression
    return booster, is_regression

eli5.xgboost._check_booster_args = _check_booster_args​
```
With the above I have `show_weights` working. I have XGBoost 0.7.post3, ELI5 0.8, Python 3.5 on Linux.

**Comment by spamula** [comment]
@sudarshan1413 @ianozsvald Passing in xgb.get_booster() into the eli5 estimator seemed to do the trick for me!

**Comment by sudarshan1413** [comment]
@ianozsvald @spamula Thanks for your comments... Let me try these out...

**Comment by ChaoXiWhite** [comment]
https://www.discoverbits.in/221/xgboost-booster-function-error-typeerror-object-callable
If you have xgboost 0.7 and sklearn 0.19 on your machine, try get_booster() instead of booster(). It will work. if it doesn't work, you need to upgrade both xgboost and sklearn.

imps = bst.get_booster().get_score(importance_type='gain')

**Comment by lopuhin** [comment]
fixed in #268
